docs/stream-aggregation.md

```diff
@@ -577,29 +579,34 @@ The results of `increase` is equal to the following [MetricsQL](https://docs.vic
 sum(increase_pure(some_counter[interval]))
 ```
 
-`increase` assumes that all the counters start from 0. For example, if the fist seen sample for new [time series](https://docs.victoriametrics.com/keyconcepts/#time-series)
+`increase` assumes that all the counters start from 0. For example, if the first seen sample for new [time series](https://docs.victoriametrics.com/keyconcepts/#time-series)
 is `10`, then `increase` assumes that the time series has been increased by `10`. If you need ignoring the first sample for new time series,
 then take a look at [increase_prometheus](#increase_prometheus).
 
 For example, see below time series produced by config with aggregation interval `1m` and `by: ["instance"]` and the regular query:
 
 <img alt="increase aggregation" src="stream-aggregation-check-increase.webp">
 
-`increase` can be used as an alternative for [rate](https://docs.victoriametrics.com/metricsql/#rate) function.
-For example, if `increase` is calculated for `some_counter` with `interval: 5m`, then `rate` can be calculated
-by dividing the resulting aggregation by `5m`:
-
-```metricsql
-some_counter:5m_increase / 5m
-```
-
-This is similar to `rate(some_counter[5m])`.
-
 Aggregating irregular and sporadic metrics (received from [Lambdas](https://aws.amazon.com/lambda/)
 or [Cloud Functions](https://cloud.google.com/functions)) can be controlled via [staleness_interval](#staleness) option.
 
 See also [increase_prometheus](#increase_prometheus) and [total](#total).
 
+### rate
+
+`rate` returns the increase of input [time series](https://docs.victoriametrics.com/keyconcepts/#time-series) over the given `interval`.
+`rate` makes sense only for aggregating [counters](https://docs.victoriametrics.com/keyconcepts/#counter).
+
+The results of `rate` is equal to the following [MetricsQL](https://docs.victoriametrics.com/metricsql/) query:
+
+```metricsql
+sum(rate(some_counter[interval]))
+```
+
+`rate` assumes that all the counters start from 0. For example, if the first seen sample for new [time series](https://docs.victoriametrics.com/keyconcepts/#time-series)
```

Comment:
From my reading of https://github.com/VictoriaMetrics/VictoriaMetrics/blob/56531abd562d9a4a4e8825d1a29d9e98c18ec7b8/app/vmselect/promql/rollup.go#L1900-L1935 the `rate` implementation doesn't assume that counter starts from 0. Is my understanding correct?

Reply:
you're right, updated it

Reply:
in this case makes sense to increase a precision of intervals or modify flushState interface. prefer first option